chain/builder.go

```diff
@@ -181,14 +181,18 @@ func BuildBlock(
 				continue
 			}
 
-			stateKeys, err := tx.StateKeys(sm)
+			stateKeysWithMode, err := tx.StateKeys(sm)
 			if err != nil {
 				// Drop bad transaction and continue
 				//
 				// This should not happen because we check this before
 				// adding a transaction to the mempool.
 				continue
 			}
+			stateKeys := set.NewSet[string](len(stateKeysWithMode))
```

Comment:
I ended up removing this and instead passed a `set[Key]` into Executor and TState so we can preserve the key mode info. this would allow the Executor to use these modes and also the TState can prefetch the keys according to the mode